src/service/public-key.js

```diff
@@ -78,8 +79,8 @@ class PublicKey {
     // if emails array is empty, all userIds of the key will be submitted
     if (emails.length) {
       // keep submitted user IDs only
-      key.userIds = key.userIds.filter(({email}) => emails.includes(email));
-      if (key.userIds.length !== emails.length) {
+      key.userIds = key.getUserIDs.filter(({email}) => emails.includes(email));
```

Comment:
Looks like this should be: `key.getUserIDs().filter`

Reply:
Thanks for the input - if you mean `key.getUserIDs().filter(({email}) => ......` that seems to work the same way, but you're right, it makes more sense with the parentheses.